Carlos/DiskCacheLevel.swift

```diff
@@ -128,7 +128,15 @@ public final class DiskCacheLevel<K: StringConvertible, T: NSCoding>: CacheLevel
   }
   
   private func pathForKey(_ key: K) -> String {
-    return (path as NSString).appendingPathComponent(key.toString().MD5String())
+    let md5PathComponent = key.toString().MD5String()
+    let strippedMd5PathComponent = stripSpecialCharactersForPath(from: md5PathComponent)
+    
+    return (path as NSString).appendingPathComponent(strippedMd5PathComponent)
+  }
+  
+  private func stripSpecialCharactersForPath(from string: String) -> String {
```

Comment:
Maybe create a string extension instead of keeping in the diskCacheLevel

Reply:
Yeah i thought about that, but since this is very related to the DiskCache I preferred to leave it there as private.

Reply:
but the stripping of special characters is pretty generic although we only need it this class. We might need it in some other place. So It might be better to just have extension method

Reply:
Even with that, the characters set to allow is specific for this task (there is no common understanding of what is a special character) so we would have to add them as a parameter. Having that, the function would be a one liner `string.filter { okayChars.contains($0) }` which makes adding it into an extension a bit overkill to me.

Reply:
okay